In [5]:
import numpy as np
from scipy.optimize import minimize

data = np.array([5.1, 5.5, 5.9, 6.0, 6.2, 5.8, 5.4, 6.1, 5.7])
n = len(data)

def neg_log_likelihood(params):
    mu, sigma = params
    if sigma <= 0:
        return np.inf
    return 0.5 * n * np.log(2 * np.pi) + n * np.log(sigma) + np.sum((data - mu)**2) / (2 * sigma**2)

initial_guess = [np.mean(data), np.mean(data)]
bounds = [(None, None), (0.01, None)]

result = minimize(neg_log_likelihood, initial_guess, method='L-BFGS-B', bounds=bounds)
mu_hat, sigma_hat = result.x
sigma2_hat = sigma_hat**2

print('Оценка среднего (mu):', mu_hat)
print('Оценка стандартного отклонения (sigma):', sigma_hat)
print('Оценка дисперсии (sigma^2):', sigma2_hat)

Оценка среднего (mu): 5.744444431895161
Оценка стандартного отклонения (sigma): 0.3370166827206089
Оценка дисперсии (sigma^2): 0.11358024443200358


Оценка среднего mu = 5.7444. Показывает «центр тяжести» данных — типичное значение выборки. То есть большинство наблюдений располагается примерно около 5.74. Оценка стандартного отклонения sigma = 0.3370 это мера разброса данных вокруг среднего. Малое значение (около 0.34) означает, что значения мало отличаются друг от друга и сгруппированы близко к среднему. Оценка дисперсии sigma^2 = 0.1136 это квадрат стандартного отклонения, ещё одна мера разброса. Она показывает, что отклонения от среднего невелики, так как дисперсия меньше 1. После оптимизации методом максимального правдоподобия мы делаем вывод, что наша выборка хорошо описывается нормальным распределением с центром около 5.74 и небольшим разбросом (sigma примерно равна 0.34). Это значит, что подавляющее большинство значений лежит примерно в диапазоне 5.4–6.1 (около одного стандартного отклонения от среднего).